data/modules/TradeShips/Debug.lua

```diff
@@ -0,0 +1,333 @@
+						end
+					}
+				})
+		end
+		if ui.collapsingHeader("Log") then
+			local obj = Game.systemView:GetSelectedObject()
+			local ship_selected = obj.type == Engine.GetEnumValue("ProjectableTypes", "OBJECT") and obj.base == Engine.GetEnumValue("ProjectableBases", "SHIP")
+			search_text, _ = ui.inputText("Search log", search_text, {})
+			drawArrayTable("tradeships_log", Core.log, Core.log.iter(search_text ~= "" and function (row)
+				return
+				row.label and string.match(row.label, search_text) or
+				row.msg and string.match(row.msg, search_text) or
+				row.model and string.match(row.model, search_text)
+			end),{
+				{ name = "Time",    key = "time" , fnc = ui.Format.Datetime                },
+				{ name = "Label",   key = "label",                           string = true },
+				{ name = "Model",   key = "model",                           string = true },
+				{ name = "Message", key = "msg",                             string = true }
+			},{
+				callbacks = {
+					onClick =	function(row)
+						status = Core.ships[row.ship] and Core.ships[row.ship].status
+						if status and status ~= "hyperspace" and status ~= "hyperspace_out" then
+							Game.systemView:SetSelectedObject(Engine.GetEnumValue("ProjectableTypes", "OBJECT"),
+								Engine.GetEnumValue("ProjectableBases", "SHIP"), row.ship)
+						end
+					end,
+					isSelected = function(row)
+						return ship_selected and obj.ref == row.ship
+					end
+			}})
+	end
+end)
+
+infosize = ui.getCursorScreenPos().y
+local obj = Game.systemView:GetSelectedObject()
+if obj.type ~= Engine.GetEnumValue("ProjectableTypes", "NONE") and Core.ships[obj.ref] then
+	local ship = obj.ref
+	local trader = Core.ships[ship]
+	if ui.collapsingHeader("Info:", {"DefaultOpen"}) then
+		property("Trader: ", ship:GetShipType() .. " " .. ship:GetLabel())
+		local status = trader.status
+		if status == "docked" then
+			status = status .. " (" .. trader.starport:GetLabel() .. ")"
+		elseif status == "inbound" then
+			local d,u = ui.Format.Distance(ship:DistanceTo(trader.starport))
+			status = status .. " (" .. trader.starport:GetLabel() .. " - " .. d .. " " ..  u .. ")"
+		end
+		property("Status: ", status)
+		if trader.fnc then
+			property("Task: ", trader.fnc .. " in " .. ui.Format.Duration(trader.delay - Game.time))
+		end
+		property("Fuel: ", string.format("%.4f", ship.fuelMassLeft) .. "/" .. ShipDef[ship.shipId].fuelTankMass .. " t")
+	end
+	if ui.collapsingHeader("Internals:", {"DefaultOpen"}) then
+		local equipItems = {}
+		local total_mass = 0
+		local equips = { "cargo", "misc", "hyperspace", "laser" }
+		for _,t in ipairs(equips) do
+			for _,et in pairs(e[t]) do
+				local count = ship:CountEquip(et)
+				if count > 0 then
+					local all_mass = count * et.capabilities.mass
+					table.insert(equipItems, {
+							name = et:GetName(),
+							eq_type = t,
+							count = count,
+							mass = et.capabilities.mass,
+							all_mass = all_mass
+						})
```

Comment:
This is doubly-indented. In general, only increase indent by one tabstop per new line, regardless of how many 'new block' characters are on the preceeding line (e.g. open-paren, open-brace, etc.)

Reply:
Done!